docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -1,638 +1,772 @@
 
-<a name="gradientvisual"></a>
-## Gradient Visual
+   NPatchVisual _nPatchVisual = null;
 
-The gradient visual renders a smooth transition of colors to the control's quad. Both linear (left in the following figure) and radial (right in the following figure) gradients are supported.
+   _nPatchVisual = new NPatchVisual();
+   _nPatchVisual.URL = _imageUrl + "heartsframe.png";
+   _nPatchVisual.RelativeSize = new RelativeVector2(1.0f, 0.3f);
+   _nPatchVisual.Origin = Visual.AlignType.BottomEnd;
+   _nPatchVisual.AnchorPoint = Visual.AlignType.BottomEnd;
 
-**Figure: Gradient visual**
+   _visualView.AddVisual("NPatch-Image1", _nPatchVisual);
 
-![Linear gradient visual](media/linear-gradient-visual.png) ![Radial gradient visual](media/radial-gradient-visual.png)
+   _nPatchVisual = new NPatchVisual();
+   _nPatchVisual.URL = _imageUrl + "heartsframe.png";
+   _nPatchVisual.RelativeSize = new RelativeVector2(0.3f, 0.6f);
+   _nPatchVisual.Origin = Visual.AlignType.TopCenter;
+   _nPatchVisual.AnchorPoint = Visual.AlignType.TopCenter;
 
-The following table lists the supported properties. The visual type is `Gradient` and the visual map is `GradientVisual`.
+   _visualView.AddVisual("NPatch-Image2", _nPatchVisual);
+   ```
 
-**Table: GradientVisualProperty properties**
+The final output after implementing the preceding code, and setting the `_visualView` background color to gray is as follows:
 
-| Property        | Type              | Required        | Description                              |
-|---------------|-----------------|---------------|----------------------------------------|
-| `StartPosition` | `Vector2`           | For linear only | The start position of the linear gradient. |
-| `EndPosition`   | `Vector2`           | For linear only | The end position of the linear gradient. |
-| `Center`        | `Vector2`           | For radial only | The center point of the radial gradient.        |
-| `Radius`        | `float`             | For radial only | The size of the radius.                  |
-| `StopOffset`    | `float` array   | No              | All the stop offsets. If not supplied, default is 0.0 and 1.0. |
-| `StopColor`     | `Vector4` array  | Yes             | The color at the stop offsets. At least two are required to show a gradient. |
-| `Units`         | `integer` or `string` | No              | The coordinate system.                   |
-| `SpreadMethod`  | `integer` or `string` | No              | Indicates what happens if a gradient starts or ends inside bounds. |
+![NPatchVisual](media/visuals/NPatchVisualExample.png)
 
-### Units
-The `Units` are used to define the coordinate system for the attributes:
- -  Start (x1, y1) and end (x2, y2) points of a line, if using a linear gradient.
- -  Center point (cx, cy) and radius (r) of a circle, if using a radial gradient.
+It is worth paying attention to the correct scaling of the N-Patch image.
```

Comment:
```suggestion 
For correct scaling of the N-Patch image, you must pay due attention.

Reply:
it was a comment to the above image. Now it does not make sense. I removed this sentence.